### ISSUE TITLE
💅 False positive for vue noVueDuplicateKeys

### ISSUE THREAD

**Issue by RodrigoSerafim** [biomejs/biome#9250]
### Environment information

vue linter rule `noVueDuplicateKeys` is throwing a false positive on prop that is used on watch.

<details>

Minimal replication vue component:
```
<template>
  <div>x</div>
</template>

<script>
	export default {
		name: 'QFormContainer',
		props: {
			modelValue: [String, Number, Object],

			formData: {
				type: Object,
				default: () => ({}),
				validator: (val) => true
			},
		},

		watch: {
			formData: {
				handler(newValue)
				{
					//do things
				},
				deep: true
			}
		}
	}
</script>
```

Rule output:
```
/main.vue:11:4 [lint/correctness/noVueDuplicateKeys](https://biomejs.dev/linter/rules/no-vue-duplicate-keys) ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Duplicate key formData found in Vue component.
  
     9 │ 			modelValue: [String, Number, Object],
    10 │ 
  > 11 │ 			formData: {
       │ 			^^^^^^^^
    12 │ 				type: Object,
    13 │ 				default: () => ({}),
  
  ℹ Key formData is also defined here.
  
    18 │ 		watch: {
  > 19 │ 			formData: {
       │ 			^^^^^^^^
    20 │ 				handler(newValue)
    21 │ 				{
```

</details>

### Rule name

noVueDuplicateKeys

### Playground link

https://biomejs.dev/playground/?lintRules=noVueDuplicateKeys&code=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8AGQAaQB2AD4AeAA8AC8AZABpAHYAPgAKADwALwB0AGUAbQBwAGwAYQB0AGUAPgAKAAoAPABzAGMAcgBpAHAAdAA%2BAAoACQBlAHgAcABvAHIAdAAgAGQAZQBmAGEAdQBsAHQAIAB7AAoACQAJAG4AYQBtAGUAOgAgACcAUQBGAG8AcgBtAEMAbwBuAHQAYQBpAG4AZQByACcALAAKAAkACQBwAHIAbwBwAHMAOgAgAHsACgAJAAkACQBtAG8AZABlAGwAVgBhAGwAdQBlADoAIABbAFMAdAByAGkAbgBnACwAIABOAHUAbQBiAGUAcgAsACAATwBiAGoAZQBjAHQAXQAsAAoACgAJAAkACQBmAG8AcgBtAEQAYQB0AGEAOgAgAHsACgAJAAkACQAJAHQAeQBwAGUAOgAgAE8AYgBqAGUAYwB0ACwACgAJAAkACQAJAGQAZQBmAGEAdQBsAHQAOgAgACgAKQAgAD0APgAgACgAewB9ACkALAAKAAkACQAJAAkAdgBhAGwAaQBkAGEAdABvAHIAOgAgACgAdgBhAGwAKQAgAD0APgAgAHQAcgB1AGUACgAJAAkACQB9ACwACgAJAAkAfQAsAAoACgAJAAkAdwBhAHQAYwBoADoAIAB7AAoACQAJAAkAZgBvAHIAbQBEAGEAdABhADoAIAB7AAoACQAJAAkACQBoAGEAbgBkAGwAZQByACgAbgBlAHcAVgBhAGwAdQBlACkACgAJAAkACQAJAHsACgAJAAkACQAJAAkALwAvAGQAbwAgAHQAaABpAG4AZwBzAAoACQAJAAkACQB9ACwACgAJAAkACQAJAGQAZQBlAHAAOgAgAHQAcgB1AGUACgAJAAkACQB9AAoACQAJAH0ACgAJAH0ACgA8AC8AcwBjAHIAaQBwAHQAPgAKAA%3D%3D&language=vue

### Expected result

This is a necessary and correct vue component definition. The rule is too wide reaching, it should not lookup duplicate keys in different sections of the component.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
